Best of its genre - Factorio Origins

Nanogamer7 opened the pull request:
Name: Nanogamer7
Suggested release date: Should be ready for issue 34

More info on discord

Nanogamer7 committed:
First draft

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+## Best of its genre - Nanogamer7
+
+We've all heard of Factorios insanely [good ratings](https://steamdb.info/stats/gameratings/) - consistently rising to second place after Portal 2 again and again.[^1] Obviously many others will try to imitate this success in the automation genre, but where did Factorio get inspired from? That's what I'll be trying to answer in this first article of hopefully a multi-part series.
```
You could add the question "was it the first game solely designed for automation or merely the first one to do it successfully?"

Nanogamer7 replied:
I hope it's ok if I'll just copy that question, it fits pretty much perfectly into my text

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Automation defined
+
+The first difficult question we'll have to answer though, would be what automation even is. In Cities:Skylines for example you are building your city, which then (theoretically) functions autonomously. The problem here is, your job isn't to automate - the cars drive themselves, once you build roads - but to manage it. In Minecraft on the other hand, you are the one to make the automation even pos ...
```
> cars drive themselves

Belts move items by themselves. There isn't much difference between placing a train in CS than in Factorio, except Factorio has much more customization. I agree City Skylines isn't automation as Factorio, but you could try to define this difference better. Maybe CS has automation more as a side-thing with more focus on aesthetics. It has very limited automation options.

Or at least try to connect CS to stuff-that-make-stuff

Nanogamer7 replied:
Hmm, good point there. I'd argue the difference between streets and belts would be at junctions, where in CS each car decides by themselves where they need to go, while in Factorio the items need to be sorted by groups/categories by a player created setup (splitters). I'll see what I can think of

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+The first difficult question we'll have to answer though, would be what automation even is. In Cities:Skylines for example you are building your city, which then (theoretically) functions autonomously. The problem here is, your job isn't to automate - the cars drive themselves, once you build roads - but to manage it. In Minecraft on the other hand, you are the one to make the automation even pos ...
+
+However that definition is still a bit vague: From what I know about e.g. Dwarf Fortress, you could consider enabling dwarfs to *make stuff* as falling under that definition. But let's not get to far from Factorio here, maybe someone else will explore this further in the future. *\*wink wink\**
```
Its sounds like youre trying to hint that DF doesnt have automation, but it does according to your definition of automation. 

And I like this reference :P I just don't think I will do it within the next month, so I don't know if we should add it. Not sure though

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### There's a mod for it
+
+Speaking of Minecraft, it's popularity boom in 2013 and with it one of the largest modding communities ever,[^2][^3] it was most definitely the first introduction to technical gameplay - and to some degree even automation - to many players, especially younger ones. But that got me thinking: With mods in the likes of Buildcraft, IndustrialCraft, and Redpower featuring a very heavy focus on industr ...
```
> Speaking of Minecraft

This sounds like you're starting a completely new point because youre done with the previous one. I would try to connect it to what you've been saying.

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+Speaking of Minecraft, it's popularity boom in 2013 and with it one of the largest modding communities ever,[^2][^3] it was most definitely the first introduction to technical gameplay - and to some degree even automation - to many players, especially younger ones. But that got me thinking: With mods in the likes of Buildcraft, IndustrialCraft, and Redpower featuring a very heavy focus on industr ...
+
+Wube's online presents, while there, is spread out across the forums, reddit, discord, and a few more, which makes it difficult to search for anything, really. But after some clever searching and sifting through a few interviews, I finally found  something: An [old forum thread](https://forums.factorio.com/viewtopic.php?f=5&t=3026) from March 2014, confirming Buildcraft (or IndustrialCraft, accor ...
```
Presence :P

heinwessels replied:
Maybe start with "i went researching.. but hard because multiple". Otherwise it seems like youre randomly telling us that Wube has a lot of social media.

heinwessels replied:
I would put a little more focus on the confirming sentence. Make it two sentences for emphasis, or give some info about the interview. I read right past the crucial piece of information.

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Great games design themselves
+
+There is a principle in game design you should follow, that goes along the lines "follow the fun".[^7] While examples for that statement often include dramatic changes like Ape Out went through,[^8] it still applies to games that don't shift that drastically in direction. Let's take Factorio itself as an example: <!-- The example should be commented by devs, and an actually relevant feature | Opt ...
```
> that you should follow

I, the reader? Or game developers?

And I think you should immediately explain what this means in one sentence or so. You name this quote, and then into an example that I didn't recognise. This makes it hard to know what you meant with "follow the fun". I only understood half-way through the paragraph.

Nanogamer7 replied:
Hmm a small problem here (I think) is, that I'm diving far to quickly far to deep into game design... I hope I managed to write an a bit easier transition now, but I feel like the fundamentally different backgrounds of the suggestions provide a huge contrast in their very nature.

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+There is a principle in game design you should follow, that goes along the lines "follow the fun".[^7] While examples for that statement often include dramatic changes like Ape Out went through,[^8] it still applies to games that don't shift that drastically in direction. Let's take Factorio itself as an example: <!-- The example should be commented by devs, and an actually relevant feature | Opt ...
+
+So once they core gameplay concept of a game is clear - which is what Factorio started out with - it's just a manner of fitting features to it. Automation? That sounds like the industrial revolution, when they invented belts. Need something to get that thing of the belts? How about inserters, they are at least not human. You get the point.
```
This pararaph is a little unclear and could use a little refining. But I see you made a comment about it :)

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Conlusio
+
+While I originally tried to find Factorios "predecessor", it stemming itself with only  the core gameplay loop as an external influence makes this conclusion even more rewarding. That being said, there have probably been more automation games in the past, and Factorio definitely had a big influence on some more recent games. Maybe we can explore this further in a future article, maybe even how Fa ...
```
> Who knows

This sounds like you're unsure if you'll write about it. I'm not sure why but it sounds a little weird. Maybe rather say something like "even though they don't like admitting it *cough*satisfactory*cough*. Or something like that.

Nanogamer7 replied:
Hmm yeah, I tried to phrase that as a sarcastic teaser (i.e. I'm definitely going to write about it), but it didn't come across right. I'm not sure what you mean with the satisfactory part, but I think I somewhat solved this in draft2

> ...maybe even how Factorio inspired more than just games.
referrence to Conors upcoming article noone knows about

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+---
+
+### Sources:
```
Personally I like simple links more in the text. I never like scrolling down to see more info, and dont like the look of a list of references in an easy-reading article. But it's totally up to you, not that big of a deal.

heinwessels reviewed (commented):
This is such a nice piece! Well formatted and thought out. And I really enjoy all.your sources, people dont put that in often. I can see youre a techincal person and most likely written a few techical things in your life. It's a cool idea, and well written. There is some feedback on where it was a little hard reading or where you can expand a little. But all in all great work!

I just hope its well researched enough to get past our fact checker though ;)

Nanogamer7 committed:
Second draft

Nanogamer7 committed:
Small change regarding DF

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Automation defined
+
+The first difficult question we'll have to answer though, would be what automation even is. In Cities:Skylines for example you are building your city, which then (theoretically) functions autonomously. The difference here is, your job isn't to automate – the cars spawn out of nothing, choose their destination, and get there by themselves, once you build roads – but to manage it. In Minecraft on t ...
```
> In Minecraft on the other hand, you are the one to make the automation even possible

I think you need to state that you're talking about vanilla here. It's still not that clear, although I'm struggling to think how to say it better. Maybe make an example, like in Factorio you use assembling machines, but in Vanilla Minecraft you need to build the assembling machine itself using various small dumb components.

Nanogamer7 replied:
Hmm I thought about adding a screenshot of a vanilla smelting array, maybe that would make it more clear (not the best approach though)

Nanogamer7 replied:
Ok I had another though about it, since I didn't mention mods earlier in the text I think it's fair to assume I'm talking about vanilla here. I could of course still add a small comment/bracket along the lines of "..., even in vanilla," or "In (vanilla) Minecraft..."

heinwessels replied:
I think it would be good enough to simply change it to 

> In Vanilla Minecraft on the other hand...

I think some people might assume that you're already talking about modded minecraft, because a lot of people think of those mods if you talk about automation in Minecraft - I think.

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+The first difficult question we'll have to answer though, would be what automation even is. In Cities:Skylines for example you are building your city, which then (theoretically) functions autonomously. The difference here is, your job isn't to automate – the cars spawn out of nothing, choose their destination, and get there by themselves, once you build roads – but to manage it. In Minecraft on t ...
+
+However that definition is still a bit vague: As someone who didn't play it, would Dwarf Fortress' gameplay of enabling dwarfs to *make stuff* would fall under my definition of automation? But let's not get to far from Factorio here, maybe someone else will explore this further in the future. *\*wink wink\**
```
You are wondering here if your theory is too vague, because you don't know if it covers DF, but then you leave it hanging. You might need to mention then " but for the purpose of this article we will assume when you make stuff that makes stuff it's automation". If it's electronic circuits, wheat and seeds, or wooden menacing spikes.

(wooden menacing spike is a memorable item from DF)

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### There's a mod for it
+
+I've already touch on Minecraft as an example for automation, and it's popularity boom in 2013 and with it one of the largest modding communities ever[^2][^3] could just have made it the most popular example of automation. But that got me thinking: With mods in the likes of Buildcraft, IndustrialCraft, and Redpower featuring a very heavy focus on industrial machinery and automation, and release d ...
```
I think you can make this step to modding more gradual. You could say something like "automation is possible in vanilla Minecraft, but it does not even begin to compare to whats possible in modded Minecraft. Since it's boom in 2013 and one of the..."

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### There's a mod for it
+
+I've already touch on Minecraft as an example for automation, and it's popularity boom in 2013 and with it one of the largest modding communities ever[^2][^3] could just have made it the most popular example of automation. But that got me thinking: With mods in the likes of Buildcraft, IndustrialCraft, and Redpower featuring a very heavy focus on industrial machinery and automation, and release d ...
```
Might be worth adding links to those mod pages. Or if the the bird is ready a gallery photo for each of the mods you mentioned. L

Nanogamer7 replied:
Problems with mod pages: The current ones only date back to 2013 and 2016 respectively. I could find an older version for BuildCraft ([link](https://sourceforge.net/projects/buildcraft/files/)), however even this one only dates back to 2011/04 and version 1.4.1

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+You might ask, why did I dive into a concept of game design now? Well I believe it is an important step on understanding how Factorio came to be. I believe, once they core gameplay concept of a game is clear – which is what Factorio started out with – it's just a manner of fitting features to it. Automation? That sounds like the industrial revolution, when they invented belts. Need something to g ...
+
+### Conclusio
```
I think you forgot somet

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Great games design themselves
+
+There is a principle in game design designers should follow, that goes along the lines "follow the fun",[^7] meaning you shouldn't try to force a game to be something it isn't, but rather develop on the ideas that are enjoyable. While examples for that statement often include dramatic changes like Ape Out went through,[^8] it still applies to games that don't shift that drastically in direction.  ...
```
You could add a little thing here about Factorio's philosophy. They're definitely following the fun, but taking great care not to give in too much. They talk a lot about how gamers will always "optimize the fun out of everything", and I'm very sure I've seen kovarex and even bilka mention this somewhere. But I really can't remember for what aspect of the game.

heinwessels commented on submissions/best-of-its-genre-nanogamer7/origins.md:
```diff
@@ -0,0 +1,46 @@
+
+### Conclusio
+
+While I originally tried to find Factorios "predecessor", it stemming itself with only  the core gameplay loop as an external influence makes this conclusion even more rewarding. That being said, there have probably been more automation games in the past, and Factorio definitely had a big influence on some more recent games. Maybe we can explore this further in a future article, maybe even how Fa ...
```
> While I originally tried to find Factorios "predecessor", it stemming itself with only the core gameplay loop as an external influence makes this conclusion even more rewarding

I'm not sure what you're trying to say here?

Nanogamer7 replied:
"predecessor" as in Infiniminer for Minecraft, Warcraft for Starcraft, etc. but Factorio doesn't have that, it only has the idea of "automation", and it not having a predecessor as descibed above feel like a more satisfying conclusion to me than just finding that predcessor.
I'm explaining this here as I have no idea how to bring this nuance between "origin" and "predecessor" in a different way to text than I already did, and was hoping you might have a better idea 😬

heinwessels reviewed (commented):
Looking good Nano! Just a little part here and there that's a little unclear, or I might be a little tired. But we should be good to go for #34 I think.

Nanogamer7 committed:
Not-a-draft 1

Resolves weasels feedback again, a few minor changes, and sources are now in there separate file (not relevant for the final article

Nanogamer7 committed:
Media

The if-gallery folder includes some images I'll only use if the gallery feature will be ready

Nanogamer7 committed:
RC2

ClaudeMetz committed:
Editing

Nanogamer7 committed:
Small Fix

Conor12345 committed:
Tech edit

Nanogamer7 committed:
Gallery caption

Nanogamer7 committed:
Small title addition

Conor12345 committed:
updates

Conor12345 committed:
Merge branch 'Factorio-Origins' of https://github.com/nanogamer7/alt-f4 into pr/Nanogamer7/473
